php/libraries/ICandidate.class.inc

```diff
@@ -0,0 +1,319 @@
+
+    /**
+     * Returns the name of the candidate's project from the
+     * config file
+     *
+     * @return string Project name
+     */
+    public function getProjectTitle(): string;
+
+    /**
+     * Returns the CandID of this candidate
+     *
+     * @return CandID 6 digit CandID
+     */
+    public function getCandID();
+
+    /**
+     * Returns the PSCID of this candidate
+     *
+     * @return string The PSCID identifier for candidate
+     */
+    public function getPSCID(): string;
+
+    /**
+     * Returns the site of this candidate
+     *
+     * @return Site
+     */
+    public function getCandidateSite(): Site;
+
+
+    /**
+     * Returns the ID of this candidate's site
+     *
+     * @return int The numeric ID of the candidate's site
+     */
+    public function getCenterID(): int;
+
+    /**
+     * Returns the date of birth of this candidate
+     *
+     * @return ?DateTime
+     */
+    public function getCandidateDoB(): ?DateTime;
+
+    /**
+     * Returns the expect date of confinement (due date)
+     * of this candidate.
+     *
+     * @return ?DateTime
+     */
+    public function getCandidateEDC(): ?DateTime;
+
+    /**
+     * Return the candidate's sex
+     *
+     * @return string|null Male|Female String describing the sex
+     *                     and null if scanner
+     */
+    public function getCandidateSex(): ?string;
+
+    /**
+     * Return whether the candidate is active
+     *
+     * @return bool
+     */
+    public function isActive(): bool;
+
+    /**
+     * Return the user that registered (created) this candidate
+     *
+     * @return string UserID of user who created this candidate.
+     */
+    public function getRegisteredBy(): string;
+
+    /**
+     * Return the name of the user account that last changed this candidate.
+     *
+     * @return User
+     */
+    public function getLastModifiedBy(): User;
+
+    /**
+     * Getter method for entityType class variable.
+     *
+     * @return string The value of entityType. 'Human' or 'Scanner'.
+     */
+    public function getEntityType(): string;
+
+    /**
+     * Returns the list of visit labels w/ corresponding SessionID as key
+     *
+     * @return array<int,string> Associative array in the format
+     *                              SessionID=>Visit_label
+     */
+    public function getListOfVisitLabels(): array;
+
+    /**
+     * Generates list of valid subprojects for a the selected Candidate's
+     * project
+     *
+     * @return array<int,int> Array of the form SubprojectID => SubprojectID
+     *
+     * @throws DatabaseException
+     */
+    public function getValidSubprojects(): array;
+
+    /**
+     * Returns the Visit_label for a subject's first TimePoint.
+     *
+     * @return string The visit label for the candidate's first visit.
+     *                  Returns an empty string if one doesn't exist.
+     */
+    public function getFirstVisit(): string;
```

Comment:
It should return a Timepoint of null

Reply:
I don't think that getTimepoints returns an chronologically ordered array. It can be but we need to specify it.